docs/tutorials/eks-tutorial.md

```diff
@@ -0,0 +1,697 @@
+
+We’ll cover:
+
+-  Tooling prerequisites
+-  VPC and networking setup
+-  Cluster bootstrapping with `eksctl`
+-  Calico installation and configuration
+-  Managed node group provisioning
+-  `urunc` installation
+-  Example deployment of unikernels
+
+## Tooling Setup for EKS Cluster Provisioning
+This section ensures your local environment is equipped with all the required tools to interact with AWS and provision your EKS cluster.
+
+### Prerequisites
+You'll need the following CLI tools installed and configured:
+
+#### 1. AWS CLI
+Used to interact with AWS services like IAM, EC2, CloudFormation, etc.
+
+```bash
+# Install AWS CLI (v2 recommended)
+curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+unzip awscliv2.zip
+sudo ./aws/install
+
+# Verify installation
+aws --version
+
+# Configure it with your credentials
+aws configure
+```
+You'll be prompted to enter:
+
+-   AWS Access Key ID 
+-   AWS Secret Access Key
+-   Default region (e.g., `eu-central-1`)
+-   Default output format (e.g., `json`)
+
+#### 2. eksctl
+The official CLI tool for managing EKS clusters.
+
+```bash
+# Download and install eksctl
+curl --silent --location "https://github.com/weaveworks/eksctl/releases/latest/download/eksctl_$(uname -s)_amd64.tar.gz" | tar xz -C /tmp
+sudo mv /tmp/eksctl /usr/local/bin
+
+# Verify
+eksctl version
+```
+#### 3. kubectl
+The Kubernetes CLI used to interact with your EKS cluster.
+```bash
+# Install kubectl (replace version as needed)
+curl -LO "https://dl.k8s.io/release/v1.30.0/bin/linux/amd64/kubectl"
+chmod +x kubectl
+sudo mv kubectl /usr/local/bin/
+
+# Verify
+kubectl version --client
+```
+
+#### 4. jq
+
+A lightweight and flexible command-line JSON processor, used in helper scripts.
+```bash
+sudo apt-get update
+sudo apt-get install -y jq
+```
+#### 5. SSH Keypair (for Node Access)
+
+Ensure you have a key pair uploaded to AWS for SSH access to EC2 instances.
+
+```bash
+# Generate an SSH key if you don’t have one
+ssh-keygen -t rsa -b 4096 -f ~/.ssh/awseks -N ""
+
+# Import the public key into AWS (or use an existing one)
+aws ec2 import-key-pair \
+  --key-name awseks \
+  --public-key-material fileb://~/.ssh/awseks.pub
+```
+## Cluster Setup
+
+We begin by provisioning an Amazon EKS cluster with private subnets and Calico as the CNI instead of the default AWS CNI.
+
+### VPC with Private Subnets
+
+We use the official EKS CloudFormation template to create a VPC with private subnets.
+
+```bash
+STACK_NAME="urunc-tutorial"
+REGION="eu-central-1"
+aws cloudformation create-stack \
+  --region $REGION \
+  --stack-name $STACK_NAME \
+  --template-url https://s3.us-west-2.amazonaws.com/amazon-eks/cloudformation/2020-10-29/amazon-eks-vpc-private-subnets.yaml
+```
+
+The output of the above command would verify the successful creation of the VPC:
+
+```console
+awsgw:~/$ ./create_vpc.sh urunc-tutorial
```

Comment:
Is this script the above command, or a different script?

Reply:
it is indeed the above command, I expanded it so that it is more clear. Will remove the line.